packages/opencensus-instrumentation-http/src/http.ts

```diff
@@ -360,12 +360,28 @@ export class HttpPlugin extends BasePlugin {
 
       const setter: HeaderSetter = {
         setHeader(name: string, value: string) {
-          request.setHeader(name, value);
+          // If outgoing request headers contain the "Expect" header, the
+          // returned ClientRequest will throw an error if any new headers are
+          // added. We need to set the header directly in the headers object
+          // which has been cloned earlier.
+          if (plugin.hasExpectHeader(options) && options.headers) {
+            options.headers[name] = value;
+          } else {
+            request.setHeader(name, value);
+          }
         }
       };
 
       const propagation = plugin.tracer.propagation;
       if (propagation) {
+        // If outgoing request headers contain the "Expect" header, the returned
+        // ClientRequest will throw an error if any new headers are added.
+        // So we need to clone the options object to be able to inject new
+        // header.
+        if (options.headers && plugin.hasExpectHeader(options)) {
```

Comment:
Is it safe to remove `options.headers` check as we are doing same in `hasExpectHeader` function?